src/main/java/parsers/CommandParser.java

```diff
@@ -0,0 +1,80 @@
+package parsers;
+
+import commands.AddTaskCommand;
+import commands.Command;
+import commands.Commands;
+import commands.DeleteTaskCommand;
+import commands.ExitCommand;
+import commands.FindTaskCommand;
+import commands.ListCommand;
+import commands.MarkTaskCommand;
+import commands.UnmarkTaskCommand;
+import exceptions.DukeException;
+import exceptions.IncompleteCommandException;
+import exceptions.UnknownCommandException;
+import formatters.StringUtils;
+
+/**
+ * Represents a parser which takes a command and outputs a Command object for Duke to execute.
+ */
+public class CommandParser extends Parser {
+
+    static final String BANNER = "____________________________________________________________";
+
+    /**
+     * Returns a Command object which can allow Duke to execute.
+     * @param response string input from stdin or command line
+     * @return Command for which Duke can execute
+     * @throws UnknownCommandException if command is not recognised by Duke
+     */
+    public Command parse(String response) throws UnknownCommandException {
+        String[] commands = response.split(" ");
+        System.out.println(BANNER);
+        try {
+            if (commands.length < 2 && isIndexRequiredCommand(commands[0])) {
+                throw new IncompleteCommandException(String.format("Hrrmmm. Not enough arguments, "
+                        + "%s has. Hmm", commands[0]), null);
+            }
+            Commands command = Commands.valueOf(commands[0].toUpperCase());
+            switch (command) {
+            case LIST:
+                return new ListCommand();
+            case MARK:
+                return new MarkTaskCommand(commands[1]);
+            case UNMARK:
+                return new UnmarkTaskCommand(commands[1]);
+            case DELETE:
+                return new DeleteTaskCommand(commands[1]);
+            case BYE:
+                return new ExitCommand();
+            case TODO:
+            case DEADLINE:
+            case EVENT:
```

Comment:
In this case, I think you can improve a third-person's understanding of the code by including a comment saying that you have purposedly allowed a fallthrough in the case statements

Reply:
Note taken! I did not think about it that way! Thank you!😊